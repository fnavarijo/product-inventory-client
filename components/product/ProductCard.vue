<template>
  <NuxtLink
    :to="`/catalog/${id}`"
    class="group m-auto cursor-pointer w-full"
  >
    <div class="relative pt-card bg-light-gray-400 rounded-md flex justify-center">
      <img
        class="object-cover absolute w-4/5 mx-auto top-1/2 transform -translate-y-1/2 group-hover:w-4/6 transition-width ease-in duration-100"
        :src="image"
        alt=""
      >
      <div class="absolute bottom-0 bg-primary-600 text-white uppercase w-full text-center p-3 opacity-0 rounded-br-md rounded-bl-md group-hover:opacity-100 transition ease-in duration-100">
        Ver
      </div>
    </div>
    <div class="flex flex-col mt-4">
      <div id="category" class="uppercase overflow-hidden">
        {{ categoryName }}
      </div>
      <!-- TODO: Not sure if the h-12 is the way to go to have default height -->
      <p class="text-lg h-12 text-dark-gray-600 font-bold line-clamped hover:text-primary-600 transition ease-in duration-100">
        {{ name }}
      </p>
    </div>
  </NuxtLink>
</template>

<script lang="ts">
import Vue from 'vue';

// TODO: remove it
// import { ProductContainerBase } from '@/components/product/ProductContainerBase';

export default Vue.extend({
  // mixins: [ProductContainerBase],
  props: {
    id: {
      type: String,
      required: true,
    },
    category: {
      type: String,
      required: true,
    },
    name: {
      type: String,
      required: true,
    },
    image: {
      type: String,
      required: true,
    },
  },
  computed: {
    categoryName (): string {
      return this.category.split('-')[0].trim();
    },
  },
});
</script>

<style scoped>
.line-clamped {
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  display: -webkit-box;
  overflow: hidden;
}
</style>
