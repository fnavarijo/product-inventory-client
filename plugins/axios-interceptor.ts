import { Context } from '@nuxt/types';

export default function ({ $axios, store }: Context): void {
  $axios.onError((error) => {
    if (error.response) {
      console.error('Response error:', error.response);
    } else if (error.request) {
      console.error('Request error:', error.request);
      store.commit('notification/setMessage', 'Algo ocurrio mientras se hacia la peticion.');
    } else {
      console.error('Something dark happened');
    }
  });
}
