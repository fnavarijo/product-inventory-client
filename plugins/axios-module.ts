import { Context } from '@nuxt/types';

import { initAxios } from '@/api';

export default ({ $axios }: Context) => {
  initAxios({ axios: $axios });
};
