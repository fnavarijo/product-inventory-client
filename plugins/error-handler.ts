import Vue from 'vue';

Vue.config.errorHandler = function () {
  return true;
};
