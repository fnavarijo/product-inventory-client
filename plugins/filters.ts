import Vue from 'vue';

export const currencyFilter = (value: string) =>
  value
    ? parseFloat(value).toFixed(2)
    : '';

Vue.filter('currency', currencyFilter);
