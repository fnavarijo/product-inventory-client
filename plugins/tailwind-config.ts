import Vue from 'vue';
// @ts-ignore
import resolveConfig from 'tailwindcss/resolveConfig';

import tailwindConfig from '@/tailwind.config';

declare module 'vue/types/vue' {
  interface Vue {
    $twConfig(): any
  }
}

Vue.prototype.$twConfig = () => resolveConfig(tailwindConfig);
