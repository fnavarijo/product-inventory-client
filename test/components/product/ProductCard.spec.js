import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils';

import ProductCard from '@/components/product/ProductCard.vue';

import { currencyFilter } from '@/plugins/filters';

const localVue = createLocalVue();
localVue.filter('currency', currencyFilter);

describe('ProductCard component', () => {
  it('should render card with name, category and price', () => {
    const wrapper = mount(ProductCard, {
      propsData: {
        category: 'Shoes',
        name: 'Nike shoes',
        price: 25.50,
        id: 1,
      },
      stubs: {
        NuxtLink: RouterLinkStub,
      },
      localVue,
    });

    expect(wrapper.text()).toContain('Shoes');
    expect(wrapper.text()).toContain('Nike shoes');
    expect(wrapper.text()).toContain('25.50');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
