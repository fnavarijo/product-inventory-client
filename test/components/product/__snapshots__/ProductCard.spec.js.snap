// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component should render card with name, category and price 1`] = `
<a class="group m-auto cursor-pointer">
  <div class="relative"><img src="~/assets/img/product.jpg" alt="" class="square-256 object-cover">
    <div class="absolute bottom-0 bg-primary-600 text-white uppercase w-full text-center p-3 opacity-0 group-hover:opacity-100 transition ease-in duration-100">
      Ver
    </div>
  </div>
  <div class="flex flex-col mt-4"><span class="uppercase font-light">Shoes</span> <span class="text-lg text-dark-gray-600 hover:text-primary-600 transition ease-in duration-100">Nike shoes</span> <span class="font-bold text-xl text-dark-gray-500">Q 25.50</span></div>
</a>
`;
